xls/contrib/integrator/area_model/area_characterization_client_main.py

```diff
@@ -0,0 +1,1001 @@
+# Lint as: python3
+#
+# Copyright 2020 The XLS Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Sweeps to characterize datapoints from a synthesis server.
+
+These datapoints can be used in an area model (where they may be interpolated)
+-- the results emitted on stdout are in xls.delay_model.DelayModel prototext
+format.
+"""
+
+import math 
+import numpy as np
+
+from typing import List, Sequence, Optional, Tuple
+
+from absl import app
+from absl import flags
+from absl import logging
+
+import grpc
+
+from xls.delay_model import delay_model
+from xls.delay_model import delay_model_pb2
+from xls.delay_model import op_module_generator
+from xls.ir.python import bits
+from xls.synthesis import client_credentials
+from xls.synthesis import synthesis_pb2
+from xls.synthesis import synthesis_service_pb2_grpc
+
+FLAGS = flags.FLAGS
+flags.DEFINE_integer('port', 10000, 'Port to connect to synthesis server on.')
+
+FREE_OPS = ('kArray kArrayConcat kBitSlice kConcat kIdentity '
+            'kLiteral kParam kReverse kTuple kTupleIndex kZeroExt kSignExt').split()
+
+# Configure regression data sweeps
+verbose=True
+
+# ceilings are exclusive
+flat_bit_count_sweep_floor=8
```

Comment:
Module scope constants should be ALL_CAPS. Also some of these constants might be better handled by encapsulating inside of a function. For example the construct:

range(flat_bit_count_sweep_floor, flat_bit_count_sweep_ceiling, base_loop_stride):

(and 2x, 3x, 4x variants) might better be handled be a function which returns the range of bit counts to sweep, maybe taking a stride factor.

Reply:
Done.